### ISSUE TITLE
Document connection pool best usage and caveats

### ISSUE THREAD

**Issue by mpaolini** [aio-libs/aiopg#129]
A wrong usage of connection pool may cause deadlocks. See #113 #127 #128 for reference. Expecially [issue 128 comment](https://github.com/aio-libs/aiopg/issues/128#issuecomment-234192294).

We should document what's the best way of using connection pooling and also give some example of bad usage that leads to deadlocks.

**Comment by mpaolini** [comment]
Also see https://github.com/KeepSafe/aiohttp/pull/981

**Comment by mpaolini** [comment]
[aiohttp demo polls app](https://github.com/KeepSafe/aiohttp/blob/master/demos/polls/aiohttpdemo_polls/views.py) has been refactored lately to match the connection pool best practices. 
I am doing similar work for `aiopg` examples too.

Basically it boils down to:

```python
async def index(request): 
     # the view function acquires the connection
    async with request.app['db'].acquire() as conn:
        # the connection is passed down to db utlity functions
        db.do_some_query(conn)
```

While the approach is sound and simple, I think it may encourage taking the connection out of the pool for longer than needed. I will point it out in the docs.

**Comment by asvetlov** [comment]
Good

**Comment by mpaolini** [comment]
Proposed patch is in #133

**Comment by asvetlov** [comment]
Fixed by #133
